packages/gestalt/src/Spinner.js

```diff
@@ -6,26 +6,35 @@ import Box from './Box.js';
 import Icon from './Icon.js';
 import styles from './Spinner.css';
 
-const SIZE = 40;
+const SIZE_NAME_TO_PIXEL = {
+  xs: 24,
+  sm: 32,
+  md: 40,
+  lg: 48,
+  xl: 56,
+};
 
 type Props = {|
   accessibilityLabel: string,
   delay?: boolean,
   show: boolean,
+  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
 |};
 
 export default function Spinner({
   accessibilityLabel,
   delay = true,
   show,
+  size = 'md',
 }: Props) {
+  const iconSize = SIZE_NAME_TO_PIXEL[size];
```

Comment:
Can we inline this directly for the prop that way if `show` is false then the browser doesnt need to compute this extraneously